spec.md

```diff
@@ -242,6 +242,29 @@ A common use of an externally defined module is to represent a Verilog module
 that will be written separately and provided together with FIRRTL-generated
 Verilog to downstream tools.
 
+## Implementation Defined Modules (Intrinsics)
+
+Intrinsic modules are modules which represent implementation-defined, 
+compiler-provided functionality.  Intrinsics generally are used for 
+functionality which requires knowledge of the implementation or circuit not
+available to a library writer.  What intrinsics are supported by an 
+implementation is defined by the implementation.  The particular intrinsic
+represented by an intrinsic module is encoded in _intrinsic_.  The name of the
+intmodule is only used for instances, it has no semantic meaning.  An 
+implementation shall type-check all ports and parameters.  Ports may be 
+uninferred (either width or reset) if specified by the implementation (which is 
+useful for inspecting and interacting with those inference features).
+
+``` firrtl
+intmodule MyIntrinsicModule :
+  input foo: UInt
+  output bar: UInt<4>
+  output baz: SInt<8>
+  intrinsic = IntrinsicName
+  parameter x = "hello"
+  parameter y = 42
```

Comment:
For clarity, it may be good to call this `intmodule IntrinsicModule_xhello_y42` or something that helps get the point across that this is a name used to uniquely identify an instantiation.